determined-ai/determined
# feat: enable token auth for Jupyter notebooks [MD-404]

## Ticket




## Description


Enables access to a remote Jupyter server running as a Determined task outside Determined (use case here is to support VSCode integrations). 

Jupyter in VSCode requires an access token, so this change modifies Determined auth logic to accept a token parameter in the proxy URLs.

## Test Plan

Testing this requires VSCode with the Jupyter extension installed.

- Launch a Jupyter notebook through the CLI:
`det notebook start`
- After the notebook starts up, it should print a URL for connecting to the running remote Jupyter server:
- Copy this URL, go to VSCode
- Create a new Jupyter notebook in VSCode
![image](https://github.com/determined-ai/determined/assets/83614683/56a33585-4598-4ea5-ba1c-e86ff674d892)
- In VSCode, click on "Select Kernel" -> "Select another kernel" -> "Existing Jupyter server"
- You should see a prompt to enter the URL of a remote Jupyter server
- Paste the URL from `det notebook start` into the prompt, it should contain a `?token=` parameter. 
- Run some commands in the local notebook shell. Verify that VSCode is able to connect to the Determined Jupyter instance. 
## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.




[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
